spring-cloud-netflix-sidecar/src/main/java/org/springframework/cloud/netflix/sidecar/SidecarConfiguration.java

```diff
@@ -80,13 +97,29 @@ public EurekaInstanceConfigBean eurekaInstanceConfigBean() {
 			int port = this.sidecarProperties.getPort();
 			config.setNonSecurePort(port);
 			config.setInstanceId(getDefaultInstanceId(this.env));
-			if(StringUtils.hasText(springAppName)) {
+			if (StringUtils.hasText(springAppName)) {
 				config.setAppname(springAppName);
 				config.setVirtualHostName(springAppName);
 				config.setSecureVirtualHostName(springAppName);
 			}
-			if (StringUtils.hasText(this.hostname)) {
-				config.setHostname(this.hostname);
+			String hostname = this.sidecarProperties.getHostname();
+			String ipAddress = this.sidecarProperties.getIpAddress();
+			if (!StringUtils.hasText(hostname) && StringUtils.hasText(this.hostname)) {
+				hostname = this.hostname;
+			}
+			if (StringUtils.hasText(hostname)) {
+				if (!StringUtils.hasText(ipAddress)) {
+					try {
+						ipAddress = inetUtils.convertAddress(InetAddress.getByName(hostname)).getIpAddress();
```

Comment:
@ryanjbaxter What can I propose: remove hostname resolution but update documentation that inform user `sidecar.ipAddress` should be fill in addition to `sidecar.hostname` to prevent any futur error when trying to use `ipAddress` from `Eureka` meta informations? Like `eureka.instance.preferIpAddress`